### PR TITLE
Fix MSVS 2019 builds

### LINKED ISSUE #867: Exception: Visual Studio Version 2015 (from GYP_MSVS_VERSION) not found.
**Operating System**: windows 10 Pro version 1909

I was using the docs and the steps to install shaka packager https://google.github.io/shaka-packager/html/build_instructions.html
I’m almost certain I did everything right. These are  the steps I did to get where I’m at.

1. I downloaded the depot_tools.zip, and extracted it inside my “c:\src\depot_tools” folder on my computer. I made sure the .git file was inside the project of depot_tools. (https://imgur.com/a/wZZzSZD)

2. Added to the PATH c:\src\depot_tools so that the executable (NOTE: I made sure this was added in front of any directory that could have python or git installed) could be found. I also added a new variable called “DEPOT_TOOLS_WIN_TOOLCHAIN” and gave it a value of “0”. I made sure that these changes were done inside the system variables, and not user variables.
(https://imgur.com/a/y8lQkOm)

3. I ran “gclient” without any arguments for the first time to install any dependencies the first time inside the “cmd” console making sure it wasn't using powershell or bash, I didn’t get any errors although I got a warning message “WARNING: Your metrics.cfg file was invalid or nonexistent. A new one will be created.”

4. Made a directory called “shaka_packager” and went inside it using this command “mkdir shaka_packager && cd shaka_packager”.

5. This is when I ran the last command “gclient sync" and got the error listed above not being able to find the VS Code version.

“Traceback (most recent call last):
  File "src/packager/tools/clang/scripts/update.py", line 921, in <module>
    sys.exit(main())
  File "src/packager/tools/clang/scripts/update.py", line 917, in main
    return UpdateClang(args)
  File "src/packager/tools/clang/scripts/update.py", line 455, in UpdateClang
    CopyDiaDllTo(os.path.join(LLVM_BUILD_DIR, 'bin'))
  File "src/packager/tools/clang/scripts/update.py", line 379, in CopyDiaDllTo
    dia_path = os.path.join(GetVSVersion().Path(), 'DIA SDK', 'bin', 'amd64')
  File "src/packager/tools/clang/scripts/update.py", line 366, in GetVSVersion
    vs_toolchain.SetEnvironmentAndGetRuntimeDllDirs()
  File "C:\Users\14068\Desktop\WebDev\boilerplates\es6\public\audio\shaka_packager\src\packager\build\vs_toolchain.py", line 78, in SetEnvironmentAndGetRuntimeDllDirs
    os.environ['GYP_MSVS_OVERRIDE_PATH'] = DetectVisualStudioPath()
  File "C:\Users\14068\Desktop\WebDev\boilerplates\es6\public\audio\shaka_packager\src\packager\build\vs_toolchain.py", line 154, in DetectVisualStudioPath
    ' not found.') % (version_as_year))
Exception: Visual Studio Version 2015 (from GYP_MSVS_VERSION) not found.
Error: Command 'vpython.bat src/packager/tools/clang/scripts/update.py --if-needed' returned non-zero exit status 1 in C:\Users\14068\Desktop\WebDev\boilerplates\es6\public\audio\shaka_packager

If you have any questions let me know. I appreciate anyone who is willing to help, I've been pulling my hair out :). Thanks again.

### PR DESCRIPTION
**Pull request by joeyparrish**
Summary of changes:

 - In order to use MSVS 2019, gyp must be upgraded.  The old version didn't recognize MSVS 2019, but the newer version does.
 - Skip clang downloads on Windows.  The older clang download tools don't recognize the MSVS 2019 environment.  We aren't using clang for a Windows build anyway, so just skip this hook when running gclient sync.
 - Because a StreamState object contains a unique_ptr, it is not copyable.  A vector of StreamStates, therefore, causes a compile error on resize or push_back, both of which invoke the copy constructor.
I don't know why MSVS complains, but clang does not. To fix this, I'm changing vector<StreamState> into deque<StreamState>.
 - The build instructions have been updated with necessary environment variables for newer MSVS versions.

At this point static_library builds are working in MSVS 2019.  shared_library builds are still not working.

Closes #867 (MSVS 2019)
Issue #318 (progress toward shared_library support on Windows)
Issue #336 (progress toward replacing Travis & Appveyor with GitHub Actions, which uses MSVS 2019)
b/190743862 (internal; tracking replacement of Travis)

### CODE REVIEW

**Comment by joeyparrish** [comment]
Unfortunately, the clang upgrade that helped fix MSVS 2019 appears to have broken macOS.  Not _quite_ ready for review after all.

**Comment by TheModMaker** [comment]
If the problem is non-movable, can't you just change from `std::vector` to `std::deque` (basically a `std::list` with random access)?

**Comment by joeyparrish** [comment]
I'll definitely try it.  Thanks!

I've got some other ideas I'd like to discuss to try to simplify the set of changes we need for this.

**Comment by joeyparrish** [comment]
deque works perfectly.  Much simpler, thanks!  I'll make that change.

**Comment by joeyparrish** [comment]
@TheModMaker helped me simplify things further.  Now all the clang-related changes are out, which is good, because the clang changes were breaking macOS.

PTAL!

File: DEPS [hunk 1]
```diff
@@ -111,6 +111,8 @@ hooks = [
   {
     # Pull clang if needed or requested via GYP_DEFINES (GYP_DEFINES="clang=1").
     "name": "clang",
+    # Skip clang updates on Windows, where we don't use clang.
+    "condition": "not checkout_win",
```
Comment (TheModMaker) [inline DEPS:115]: Did you double-check this does run this on Linux/Mac?  Just making sure this isn't being ignored and not run anywhere.

**Review by TheModMaker — APPROVED** [review]

File: DEPS (hunk 1 above)
Comment (joeyparrish) [reply DEPS:115]: I just checked, and it still runs the hook on Linux.

**Pull request merged**